### ISSUE TITLE
🎯 Systematic Hyperparameter Optimization

### ISSUE THREAD

**Issue by SergioSV96** [SergioSV96/multiagent#3]
## Problem

The project has a basic Optuna-based hyperparameter tuning example (`configs/hyperparameter_tuning.yaml`), but there is no systematic approach to finding good hyperparameters for different map/algorithm/reward combinations. This likely contributes to training instability and the asymmetric learning problem (one agent learns while the other does not).

## Current State

- `configs/hyperparameter_tuning.yaml` exists with 50 Optuna trials on oval_small + PPO
- Tuned params: gamma, lr, clip_param, train_batch_size, num_sgd_iter, vf_clip_param
- All 24 experiments in `experiments.yaml` use **identical default hyperparameters** regardless of map complexity or reward function
- No documented "good" hyperparameter sets for any configuration
- HP tuning code exists but analysis notebooks have commented-out plotting

### 1. Map-Specific Hyperparameter Defaults

Run targeted HP sweeps for each map + algorithm combination and document the best defaults:

| Map | Algorithm | Key Differences to Tune |
|-----|-----------|----------------------|
| oval_small (easy) | PPO/SAC | Standard params likely work |
| Catalunya (hard, many turns) | PPO/SAC | May need smaller lr, larger networks, more timesteps |
| Spielberg (medium) | PPO/SAC | Intermediate tuning |
| Monza (high speed) | PPO/SAC | May need different clip_param for stability at high speeds |

### 2. Reward Function Sensitivity Analysis

Different reward functions have different optimization landscapes:
- `ProgressRewardAdvancedEnv` — sparse rewards, may need larger batch size
- `SpeedRewardEnv` — continuous rewards, may converge faster with smaller lr
- `WaypointRewardEnv` — structured guidance, may work with simpler networks
- `SafetyRewardEnv` — LiDAR-based, may benefit from different network architecture

### 3. Automated HP Tuning Pipeline

Create a script that:
1. Takes an experiment config as input
2. Runs OptunaSearch with configurable trial count
3. Saves best hyperparameters to a new YAML file (`configs/hp_defaults.yaml`)
4. Generates a summary report of top configurations

```bash
# Run HP tuning for a specific experiment
python run.py tune --experiment oval_small_PPO_ProgressRewardAdvancedEnv --trials 50

# Auto-generate defaults for all experiments
python scripts/auto_tune_all.py --trials-per-exp 20
```

### 4. Document Best Practices

Create `configs/hp_defaults.yaml` with recommended hyperparameters per combination:

```yaml
hp_defaults:
  oval_small_PPO_ProgressRewardAdvancedEnv:
    lr: 1e-4
    train_batch_size: 512
    num_sgd_iter: 7
    ...
  Catalunya_PPO_WaypointRewardEnv:
    lr: 5e-5
    train_batch_size: 1024
    ...
```

## Acceptance Criteria

- [ ] HP defaults documented per map + algorithm combination in a YAML file
- [ ] `run.py tune` subcommand for running OptunaSearch from CLI
- [ ] Reward function sensitivity analysis completed (at least for PPO on oval_small)
- [ ] Best hyperparameters identified and saved to configs/hp_defaults.yaml
- [ ] Documentation of which params matter most per configuration
